api/envoy/api/v2/route/route_components.proto

```diff
@@ -875,6 +875,16 @@ message RouteAction {
 
   InternalRedirectAction internal_redirect_action = 26;
 
+  // An internal redirect is handled, iff the number of previous internal redirects that a
```

Comment:
Can you talk a bit about the motivation for this? I understand that bad configuration could lead to an infinite redirect loop, for example, but is this a practical concern? Can we just reject config that has redirect loops?

Reply:
So what I really need for my project is to make Envoy follow redirects more than 1 hop. Our setup would have a first redirect from one route to a second route, and 0-n (n <= 3) redirects within the second route.

What I am trying to capture with this config is the case for the 0-n redirects on the same route.

I do think having a way to express loop prevention is a great idea and allowing only self loop for n hops.

Is this something I should pursue?

Reply:
This addresses the motivation question, thanks.